Software/Python/easygopigo.py

```diff
@@ -46,56 +46,33 @@ def debug(in_str):
         print(in_str)
 
 
-def _grab_read():
-    '''
-    first acquire at the process level,
-    then at the thread level.
-    '''
-    global read_is_open
-    # print("acquiring")
-    try:
-        I2C_Mutex_Acquire()
-    except Exception as e:
-        print("_grab_read: {}".format(e))
-        pass
-    # while read_is_open is False:
-    #     time.sleep(0.01)
-    read_is_open = False
-    # print("acquired")
-
-
-def _release_read():
-    global read_is_open
-    try:
-        I2C_Mutex_Release()
-    except Exception as e:
-        print("_release_read: {}".format(e))
-        pass
-    read_is_open = True
-    # print("released")
-    
-
 class EasyGoPiGo():
     '''
     Wrapper to access the gopigo functionality with mutex in place
     this makes the gopigo thread safe and process safe
     if mutex is not available, then it's just a direct access to gopigo
     '''
-    def __init__(self):
+    def __init__(self, use_mutex = False):
         '''
-        On Init, set speed to half-way, so GoPiGo is predictable 
+        On Init, set speed to half-way, so GoPiGo is predictable
             and not too fast.
         '''
-        DEFAULT_SPEED = 128
-        gopigo.set_speed(DEFAULT_SPEED)
-        
+        self.DEFAULT_SPEED = 128
+        gopigo.set_speed(self.DEFAULT_SPEED)
+
+        self.mutex = None
+        if use_mutex is True:
+            self.mutex = Mutex()
+
     def volt(self):
-        _grab_read()
+        if self.mutex:
```

Comment:
How about having mutex.acquire() and mutex.release()  receive a Bool argument. If False, they just do nothing. At least easygopigo will be be burdened by twice the number of lines.  There are other approaches too that won't touch the mutex class, like creating a base method similar to _grab_read/release_read.